File: day01/solution.py
depths = [ int(l) for l in open('input.txt') ]

increased = 0
for i in range(1, len(depths)):
    if depths[i] > depths[i-1]:
        increased += 1

print('Part 1', increased)

increased = 0
window = depths[0] + depths[1] + depths[2]
for i in range(3, len(depths)):
    new_window = window - depths[i-3] + depths[i]
    if new_window > window:
        increased += 1
    window = new_window

print('Part 2', increased)

File: day01/solution.c
#include <stdio.h>
#include <stdlib.h>

#define BUFFER_LEN 500
#define MAX_DEPTHS 2000

int main(void)
{
    char buffer[BUFFER_LEN];
    FILE *fp = fopen("input.txt", "r");
    int depths[MAX_DEPTHS];
    int ndepths = 0;

    while (fgets(buffer, BUFFER_LEN, fp) != NULL)
    {
        depths[ndepths] = atoi(buffer);
        ndepths++;
    }

    fclose(fp);

    int increased = 0;
    for (int i = 1; i < ndepths; i++)
    {
        if (depths[i] > depths[i-1])
            increased++;
    }
    printf("Part 1 : %d\n", increased);

    increased = 0;
    int window = depths[0] + depths[1] + depths[2];
    for (int i = 3; i < ndepths; i++)
    {
        int new_window = window - depths[i-3] + depths[i];
        if (new_window > window)
            increased++;
        window = new_window;
    }
    printf("Part 2 : %d\n", increased);
}
